File: BMICalculator/app/src/main/java/com/s15851/bmi_calc/BmiActivity.java
package com.s15851.bmi_calc;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class BmiActivity extends AppCompatActivity {

    private EditText weight;
    private EditText height;
    private TextView bmi;
    private TextView bmiResult;
    private String BMI;
    static String bmiValueString;

    public void setBMI(String BMI) {
        this.BMI = BMI;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmi);


        Button btLaunchBmrActivity = findViewById(R.id.launchBmr);
        Button btLaunchDietActivity = findViewById(R.id.launchDiet);
        Button btLaunchQuizActivity = findViewById(R.id.launchQuiz);
        Button btLaunchStatisticsActivity = findViewById(R.id.launchStatistics);

        btLaunchStatisticsActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), StatisticsActivity.class);
                startActivity(intent);
            }
        });

        btLaunchQuizActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), QuizActivity.class);
                startActivity(intent);
            }
        });

        btLaunchBmrActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), BmrActivity.class);
                startActivity(intent);
            }
        });

        btLaunchDietActivity.setOnClickListener(new View.OnClickListener() {
            @Override
//            public void onClick(View view) {
//                Intent intent = new Intent(BmiActivity.this, DietActivity.class);
//                intent.putExtra("bmiResult", bmiValueString);
//                if (!bmiValueString.isEmpty())
//                {
//                    startActivity(intent);
//                }
//            }
            public void onClick(View view) {
                Intent intent = new Intent(BmiActivity.this, DietActivity.class);
                intent.putExtra("bmiResult", bmiValueString);
                if (bmiValueString==null || bmiValueString.isEmpty()) {
                    alert();
                } else {
                    startActivity(intent);
                }
            }
        });

        weight = findViewById(R.id.weight);
        height = findViewById(R.id.height);
        bmi = findViewById(R.id.bmi);
        bmiResult = findViewById(R.id.bmiResult);
        Button btCalculateBmi = findViewById(R.id.calcBmi);

        btCalculateBmi.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculateBmi();
            }
        });
    }

    public String calculateBmi() {
        String inputWeight = weight.getText().toString();
        String inputHeight = height.getText().toString();

        if (!inputHeight.isEmpty() && !inputWeight.isEmpty()) {
            double weightValue = Double.parseDouble(inputWeight);
            double heightValue = Double.parseDouble(inputHeight);
            double bmiValue = (double) (Math.round(weightValue / (heightValue * heightValue) * 10000));

            bmiValueString = Double.toString(bmiValue);

            bmi.setText(bmiValueString);

            if (bmiValue <= 18.5) {
                bmiValueString = getString(R.string.under);
            } else if (bmiValue > 18.5 && bmiValue < 25) {
                bmiValueString = getString(R.string.normal);
            } else if (bmiValue >= 25 && bmiValue < 30) {
                bmiValueString = getString(R.string.over);
            } else {
                bmiValueString = getString(R.string.obese);
            }
            bmiResult.setText(bmiValueString);
            BMI = bmiValueString;
            setBMI(BMI);

        }
        return bmiValueString;
    }

    private void alert() {
        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(BmiActivity.this);
        alertDialogBuilder
                .setMessage("To launch recommended diet, You must calculate your BMI first")
                .setCancelable(false)
                .setNegativeButton("Back",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                startActivity(new Intent(getApplicationContext(), BmiActivity.class));
                                finish();
                            }
                        });
        AlertDialog alertDialog = alertDialogBuilder.create();
        alertDialog.show();
    }
}

File: BMICalculator_Kotlin/app/src/main/java/com/s15851/bmicalculator/BmiActivity.kt
package com.s15851.bmicalculator

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_bmi.*


class BmiActivity : AppCompatActivity() {

    companion object {
        var bmiValueString = ""
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmi)

        calcBmi.setOnClickListener { calculateBmi() }

        launchStatistics.setOnClickListener {
            val intent = Intent(this, StatisticsActivity::class.java)
            startActivity(intent)
        }
//        launchQuiz.setOnClickListener {
//            val intent = Intent(this, QuizActivity::class.java)
//            startActivity(intent)
//        }
        launchBmr.setOnClickListener {
            val intent = Intent(this, BmrActivity::class.java)
            startActivity(intent)
        }
        launchDiet.setOnClickListener {
            val intent = Intent(this, DietActivity::class.java)
            intent.putExtra("bmiResult", bmiValueString)
            if (bmiValueString.isEmpty()) {
                alert()
            } else {
                startActivity(intent)
            }
        }
    }

    private fun calculateBmi(): String {

        val inputWeight = weight.text.toString()
        val inputHeight = height.text.toString()

        if (inputHeight.isNotEmpty() && inputWeight.isNotEmpty()) {
            val heightValue = inputHeight.toDouble()
            val weightValue = inputWeight.toDouble()
            val bmiValue = (weightValue / (heightValue * heightValue) * 10000)
            bmi.text = "%.2f".format(bmiValue)


            fun bmiResult(bmiValue: Double) : String {
                if (bmiValue <= 18.5) {
                    bmiValueString = getString(R.string.under)
                } else if (bmiValue > 18.5 && bmiValue < 25) {
                    bmiValueString = getString(R.string.normal)
                } else if (bmiValue >= 25 && bmiValue < 30) {
                    bmiValueString = getString(R.string.over)
                } else {
                    bmiValueString = getString(R.string.obese)
                }
                return bmiValueString
            }

            bmiResult.text = bmiResult(bmiValue)
        }
        return bmiValueString
    }

    private fun alert() {
        val builder = AlertDialog.Builder(this@BmiActivity)
        builder.setMessage("To launch recommended diet, You must calculate your BMI first")
        builder.setCancelable(false)
        builder.setNegativeButton("Back") { _,_ ->
                startActivity(Intent(this, BmiActivity::class.java))
                finish()
            }
        val alertDialog = builder.create()
        alertDialog.show()
    }
}